rust/src/go/nmstate/nmstate_test.go

```diff
@@ -0,0 +1,16 @@
+package nmstate
+
+import "testing"
+import "github.com/stretchr/testify/assert"
+
+func TestRetrieveNetState(t *testing.T) {
+	netState, err := RetrieveNetState(None)
+	assert.NoError(t, err, "must succeed calling retrieve_net_state c binding")
+	assert.NotEmpty(t, netState, "net state should not be empty")
+}
+
+func TestRetrieveNetStateKernelOnly(t *testing.T) {
+	netState, err := RetrieveNetState(KernelOnly)
+	assert.NoError(t, err, "must succeed calling retrieve_net_state c binding")
+	assert.NotEmpty(t, netState, "net state should not be empty")
+}
```

Comment:
Don't we want to test the rest of them ?

Reply:
Yes, I will create test for the other functions in a follow up PR. I just wanted to introduce very basic test coverage